flutter_searchbox_ui/lib/src/components/selected_filters.dart

```diff
@@ -0,0 +1,363 @@
+  ///          onDeleted: () {
+  ///            options.clearValue(id);
+  ///          },
+  ///        ),
+  ///      );
+  ///    });
+  ///    return Wrap(
+  ///      spacing: 16.0,
+  ///     crossAxisAlignment: WrapCrossAlignment.start,
+  ///     children: widgets,
+  ///    );
+  ///   },
+  ///   ...
+  /// )
+  /// ```
+
+  final Widget Function([BuildOptions? options])? buildFilters;
+  SelectedFilters({
+    this.subscribeTo,
+    this.filterLabel,
+    this.showClearAll = true,
+    this.clearAllLabel,
+    this.onClearAll,
+    this.onClear,
+    this.resetToDefault,
+    this.defaultValues,
+    this.hideDefaultValues,
+    this.buildFilters,
+    Key? key,
+  }) : super(key: key) {
+    if (resetToDefault != null || hideDefaultValues != null) {
+      if (defaultValues == null) {
+        throw ("defaultValues property is required, when resetToDefault/ hideDefaultValues properties are used.");
+      }
+    }
+  }
+
+  @override
+  _SelectedFiltersState createState() => _SelectedFiltersState();
+}
+
+class _SelectedFiltersState extends State<SelectedFilters> {
+  final Map<String, dynamic> _selectedFilters = {};
+  @override
+  initState() {
+    super.initState();
+
+    WidgetsBinding.instance?.addPostFrameCallback((_) => setSelectedFilters());
+  }
+
+  Map<String, SearchController> get activeWidgets {
+    return SearchBaseProvider.of(context).getActiveWidgets();
+  }
+
+  dynamic getResetValue(String id) {
+    if (widget.resetToDefault == true) {
+      return widget.defaultValues![id] ?? null;
+    }
+
+    return null;
+  }
+
+  void setSelectedFilters() {
+    final activeWidgets = this.activeWidgets;
+    for (var id in activeWidgets.keys) {
+      if (widget.subscribeTo != null && widget.subscribeTo!.isNotEmpty) {
+        if (!widget.subscribeTo!.contains(id)) {
+          continue;
+        }
+      }
+      final componentInstance = activeWidgets[id];
+      componentInstance?.subscribeToStateChanges((changes) {
+        setState(() {
+          final currentValue = changes['value']?.next;
+          if (currentValue != "" &&
+              currentValue != null &&
+              ((currentValue is Map || currentValue is List)
+                  ? currentValue.isNotEmpty
+                  : false) &&
+              (widget.hideDefaultValues == true
+                  ? !isEqual(currentValue, widget.defaultValues![id])
+                  : true)) {
+            _selectedFilters[id] = currentValue;
+          } else {
+            _selectedFilters.remove(id);
+          }
+        });
+      }, ["value"]);
+    }
+  }
+
+  void clearFilter(String id, [dynamic resetTo]) {
+    final activeWidgets = this.activeWidgets;
+    if (widget.onClear != null) {
+      widget.onClear!(id, activeWidgets[id]?.value);
+    }
+
+    _selectedFilters.remove(id);
+    activeWidgets[id]?.setValue(resetTo ?? getResetValue(id),
+        options: Options(
+          triggerCustomQuery: true,
+          triggerDefaultQuery: true,
```

Comment:
Is there a reason to trigger the default query?

Reply:
solved